src/ProducerEventProcessor/ProducerEventProcessorTest.cpp

```diff
@@ -1321,33 +1324,189 @@ TEST(ProducerEventProcessor, ThreadNameSmoke) {
+      thread_state_slice_producer_event1.mutable_thread_state_slice();
+  thread_state_slice1->set_pid(kPid1);
+  thread_state_slice1->set_tid(kTid1);
+  thread_state_slice1->set_thread_state(ThreadStateSlice::kRunnable);
+  thread_state_slice1->set_duration_ns(kDurationNs1);
+  thread_state_slice1->set_end_timestamp_ns(kTimestampNs1);
+  thread_state_slice1->set_switch_out_or_wakeup_callstack_status(
+      ThreadStateSlice::kWaitingForCallstack);
+  thread_state_slice1->set_switch_out_or_wakeup_callstack_id(0);
+  ThreadStateSlice expected_thread_state_slice1 = *thread_state_slice1;
+
+  ProducerCaptureEvent thread_state_slice_callstack_producer_event2;
+  ThreadStateSliceCallstack* thread_state_slice_callstack2 =
+      thread_state_slice_callstack_producer_event2.mutable_thread_state_slice_callstack();
+  thread_state_slice_callstack2->set_thread_state_slice_tid(kTid2);
+  thread_state_slice_callstack2->set_timestamp_ns(kTimestampNs2 - kDurationNs2);
+  Callstack* callstack2 = thread_state_slice_callstack2->mutable_callstack();
+  callstack2->add_pcs(kCallstackFrame1);
+  callstack2->add_pcs(kCallstackFrame2);
+  callstack2->add_pcs(kCallstackFrame3);
+
+  ProducerCaptureEvent thread_state_slice_producer_event2;
+  ThreadStateSlice* thread_state_slice2 =
+      thread_state_slice_producer_event2.mutable_thread_state_slice();
+  thread_state_slice2->set_pid(kPid1);
+  thread_state_slice2->set_tid(kTid2);
+  thread_state_slice2->set_thread_state(ThreadStateSlice::kRunnable);
+  thread_state_slice2->set_duration_ns(kDurationNs2);
+  thread_state_slice2->set_end_timestamp_ns(kTimestampNs2);
+  thread_state_slice2->set_switch_out_or_wakeup_callstack_status(
+      ThreadStateSlice::kWaitingForCallstack);
+  thread_state_slice2->set_switch_out_or_wakeup_callstack_id(0);
+  ThreadStateSlice expected_thread_state_slice2 = *thread_state_slice2;
+
+  std::vector<ClientCaptureEvent> actual_client_capture_events;
+
+  uint64_t actual_callstack_key = 0;
+  EXPECT_CALL(collector, AddEvent)
+      .Times(3)
+      .WillRepeatedly(Invoke([&actual_client_capture_events,
+                              &actual_callstack_key](ClientCaptureEvent&& client_capture_event) {
+        if (client_capture_event.has_interned_callstack()) {
+          ASSERT_EQ(actual_callstack_key, 0);
+          actual_callstack_key = client_capture_event.interned_callstack().key();
+        }
+        actual_client_capture_events.push_back(std::move(client_capture_event));
+      }));
+
+  producer_event_processor->ProcessEvent(1,
+                                         std::move(thread_state_slice_callstack_producer_event1));
+  producer_event_processor->ProcessEvent(1, std::move(thread_state_slice_producer_event1));
+  producer_event_processor->ProcessEvent(1,
+                                         std::move(thread_state_slice_callstack_producer_event2));
+  producer_event_processor->ProcessEvent(1, std::move(thread_state_slice_producer_event2));
```

Comment:
You can just use `kLinuxTracingProducerId` instead of the `1`s.

Reply:
Done.